extensions/common/context.cc

```diff
@@ -551,5 +550,21 @@ bool sanitizeBytes(std::string* buf) {
   return modified;
 }
 
+// Used for `destination_service` fallback. Unlike elsewhere when that fallback
+// to workload name, this falls back to "unknown" when the canonical name label
+// is not found. This preserves the existing behavior for `destination_service`
+// labeling. Using a workload name as a service name could be potentially
+// problematic.
+std::string getServiceNameFallback() {
```

Comment:
This will happen on both inbound and outbound right? I think we only want this fallback on inbound.

Reply:
+1. It's not correct for outbound (although gateways are outbound).

Reply:
good point. handling that will require more grace.

Reply:
fixed.